version: '3.8'

networks:
  gps-network:
    driver: bridge

services:
  db:
    image: mysql:8
    container_name: mysql_db
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: gps
      MYSQL_USER: admin
      MYSQL_PASSWORD: Utalca_123
    ports:
      - "3306:3306"
    volumes:
      - db_data:/var/lib/mysql
    restart: always
    networks:
      - gps-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "admin", "-pUtalca_123"]
      timeout: 20s
      retries: 10

  backend:
    build: ./backend
    ports:
      - "3001:3001"
    environment:
      - NODE_ENV=production
      - JWT_SECRET=claveSecreta
    depends_on:
      db:
        condition: service_healthy
    restart: always
    networks:
      - gps-network

  frontend:
    build: ./frontend
    ports:
      - "3000:80"
    depends_on:
      - backend
    restart: always
    networks:
      - gps-network

  adminer:
    image: adminer
    container_name: adminer_web
    restart: always
    ports:
      - 8080:8080
    depends_on:
      - db
    networks:
      - gps-network

volumes:
  db_data:
